### PR TITLE
bluetooth: host: Some small CS fixes and improvements

### PR DESCRIPTION
**Pull request by olivier-le-sage**
1. Add docs on valid t_XX times for CS Test command
2. Fix values for minimizing/maximizing TX power
3. Add defines for RTT packet quality and RSSI for mode 1 steps
4. Make ToX-ToX values int16_t so the user doesn't need to cast
5. minor doc fix for bt_le_cs_step_data_parse()

### CODE REVIEW

**Comment by jhedberg** [comment]
@olivier-le-sage when you have to enumerate multiple independent things, even though they're part of the same area/feature, it's a good indication that you might want to split the changes up into individual commits. In case there's a bug in one of the changes this then helps with bisectability. You also get a clearer mapping from the commit message to the actual change.

**Comment by olivier-le-sage** [comment]
> @olivier-le-sage when you have to enumerate multiple independent things, even though they're part of the same area/feature, it's a good indication that you might want to split the changes up into individual commits. In case there's a bug in one of the changes this then helps with bisectability. You also get a clearer mapping from the commit message to the actual change.

sure, can split it up.

**Review by theob-pro — DISMISSED** [review]

File: include/zephyr/bluetooth/cs.h [hunk 1]
```diff
@@ -540,6 +590,35 @@ struct bt_le_cs_subevent_step {
...
+ *
+ * @return Real and imaginary terms as int16_t.
+ */
+static inline struct bt_le_cs_iq_sample bt_le_cs_parse_pct(const uint8_t pct[3])
+{
+	uint32_t sample = sys_get_le24(pct);
+	int16_t i;
+	int16_t q;
+	struct {
+		int16_t x: 12;
```
[the remark this answers is not in the document]
Comment (olivier-le-sage) [reply include/zephyr/bluetooth/cs.h:615]: an alternative is maybe

int16_t i = ((sample & BT_HCI_LE_CS_PCT_MASK) ^ BIT(11)) - BIT(11);
int16_t q = ((sample >> 12) ^ BIT(11)) - BIT(11);

File: include/zephyr/bluetooth/cs.h [hunk 2]
```diff
@@ -540,6 +590,35 @@ struct bt_le_cs_subevent_step {
...
+
+/** @brief Extract IQ sample from HCI-formatted PCT.
+ *
+ * Convenience function for processing 24-bit phase correction terms found
+ * in CS step data. The 12-bit signed real and imaginary components are
+ * converted to host endianness and sign-extended.
+ *
+ * @param pct 24-bit little-endian phase correction term.
+ *
+ * @return Real and imaginary terms as int16_t.
```
Comment (jhedberg) [inline include/zephyr/bluetooth/cs.h:607]: It's not returning `int16_t` though, rather `struct bt_le_cs_iq_sample`.

File: include/zephyr/bluetooth/cs.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhedberg) [reply include/zephyr/bluetooth/cs.h:615]: Both look quite hard to parse for me, however I'd at least add a code comment with a reference to the sign extension web page.

File: include/zephyr/bluetooth/cs.h (hunk 2 above)
Comment (jhedberg) [reply include/zephyr/bluetooth/cs.h:607]: > It's not returning `int16_t` though, rather `struct bt_le_cs_iq_sample`.

Nevermind - it's just a reference to the implementation of `structure bt_le_cs_iq_sample`. It's just not clear from the text alone that it contains two `int16_t` parts.

Comment (olivier-le-sage) [reply include/zephyr/bluetooth/cs.h:607]: can rephrase to `@return bt_le_cs_iq_sample struct containing real and imaginary terms as int16_t`

Comment (jhedberg) [reply include/zephyr/bluetooth/cs.h:607]: Sure

File: include/zephyr/bluetooth/cs.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (alwa-nordic) [reply include/zephyr/bluetooth/cs.h:615]: Assigning a value to a signed integer variable that is larger than the variable can represent is technically implementation specific behavior in C.

It's probably OK and left implementation specific just in case the code runs a machine with non-twos complement integers. Maybe we can get a guarantee from Zephyr toolchain maintainers that this trick is supported?

Still, performing the conversion arithmetically avoids this caveat. I think it can be make quite clean:

```c
struct bt_le_cs_iq_sample bt_le_cs_parse_pct(const uint8_t pct[3])
{
    uint32_t sample = sys_get_le24(pct);

    /* Exract fields i and q. */
    uint16_t i2 = sample % (1<<12);
    uint16_t q2 = sample / (1<<12);

    /* Decode the fields from twos complement (bit width 12) into signed integers. */
    int16_t i = (i2 ^ (1<<11)) - (1<<11);
    int16_t q = (q2 ^ (1<<11)) - (1<<11);

    return (struct bt_le_cs_iq_sample){.i = i, .q = q};
}
```

[the remark this answers is not in the document]
Comment (olivier-le-sage) [reply include/zephyr/bluetooth/cs.h]: decided to introduce this later in a separate PR

**Review by jhedberg — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]
